onnxruntime/test/shared_lib/custom_op_utils.h

```diff
@@ -151,6 +151,34 @@ struct MyCustomOpWithOptionalInput : Ort::CustomOpBase<MyCustomOpWithOptionalInp
   const char* provider_;
 };
 
+struct MyCustomKernelWithVariadicIO {
+  explicit MyCustomKernelWithVariadicIO(const OrtKernelInfo* info);
+
+  void Compute(OrtKernelContext* context);
+
+ private:
+  int64_t num_inputs_;
+};
+
+struct MyCustomOpWithVariadicIO : Ort::CustomOpBase<MyCustomOpWithVariadicIO, MyCustomKernelWithVariadicIO> {
+  MyCustomOpWithVariadicIO() = default;
+
+  void* CreateKernel(const OrtApi& /* api */, const OrtKernelInfo* info) const { return new MyCustomKernelWithVariadicIO(info); };
+  const char* GetName() const { return "VariadicNode"; }
+
+  size_t GetInputTypeCount() const { return 1; };
+  ONNXTensorElementDataType GetInputType(size_t /*index*/) const { return ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING; };
+  OrtCustomOpInputOutputCharacteristic GetInputCharacteristic(size_t /* index */) const {
+    return OrtCustomOpInputOutputCharacteristic::INPUT_OUTPUT_VARIADIC;
```

Comment:
I think the answer to the question is: it depends. Each time the custom op code would represent a different Kernel  wrapping a different model, as it were and it may or may not be VARIADIC depending on the context.

Reply:
I created and merged a separate PR that added and documented variadic inputs/outputs for custom ops: https://github.com/microsoft/onnxruntime/pull/13946